misc/pstring.cpp

```diff
@@ -77,3 +79,32 @@ std::size_t CleanupStr(char *dest, const char *src, std::size_t destlen) {
 
   return out_size;
 }
+
+std::string StringJoin(const std::vector<std::string> &strs, const std::string &delim) {
```

Comment:
This can probably be simplified using [ostream_iterator](https://en.cppreference.com/w/cpp/iterator/ostream_iterator)